### DISCUSSION TITLE
Trying to make an axis-dependent zoom effect work

### DISCUSSION THREAD

**Discussion by amkram** [visgl/deck.gl#6513] [Q&A]
I've been struggling to get a specific zoom effect working on a simple example as shown [here](https://replit.com/@AlexanderKrame2/TestingDeckGLZoom#src/App.jsx). I'm using the newly added feature as of 8.6 to allow independent zooming of X and Y axes in `OrthographicView`.
I would like to be able to scroll in the left half of the screen to zoom the left scatterplot vertically along the y-axis, and the right-hand scatterplot should remain aligned vertically during this. This is working as intended. I also would like scrolling on the right side of the screen to expand/contract the right-hand scatterplot horizontally without affecting the left-hand zoom. This is also working. However, once the right-hand plot has been zoomed, it should remain at its horizontal zoom level if the left-hand plot is subsequently zoomed in/out. Currently, zooming on the left side will reset the right-hand side’s zoom level. I imagine the answer lies somewhere in the `onViewStateChange` function, maintaining the target/zoom arrays of each view, but I haven’t been able to find a solution yet. I would greatly appreciate any advice!

**Comment by Pessimistress** [comment]
You can lock view b's x zoom like so:

```js
onViewStateChange={({viewId, viewState}) => {
  if (viewId == "a") {
    //Left view changed
    setViewStates({
      a: viewState,
      b: {
        ...viewState,
        zoom: [viewStates.b.zoom[0], viewState.zoom[1]]
      }
    });
  }
...
}}
```

**Comment by amkram** [comment]
Thanks for the help. I haven't yet been able to achieve perfect alignment between both sides, so I'm using a different strategy for now. Thanks!

**Reply by charlieforward9** [reply]
its been a bit - but if you had that snippet laying around - this discussion can be closed - im doing a big bash for the community.
